core/events/notifications/Email.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Gaia\Events\Notification;
+
+use Phalcon\Events\Event;
+use PHPMailer\PHPMailer\PHPMailer;
+use PHPMailer\PHPMailer\SMTP;
+
+/**
+ * This class is used to send email notifications.
+ *
+ * @author   Rana Nouman <ranamnouman@gmail.com>
+ * @package  Foundation
+ * @category Events
+ * @license  http://www.gnu.org/licenses/agpl.html AGPLv3
+ */
+class Email
+{
+    /**
+     * This function is used to send the password reset link to the user.
+     *
+     * @method resetPassword
+     * @param  Event                 $event
+     * @param  \Gaia\MVC\Models\User $user
+     * @return void
+     * @throws \Exception
+     */
+    public function resetPassword(Event $event, $user)
+    {
+        try {
+            $di = \Phalcon\Di::getDefault();
+            $smtpConfig = $di->get('config')->get('smtp');
+            $request = \OAuth2\Request::createFromGlobals();
+            $host = getenv('FRONTEND_HOST');
+            $resetToken = $this->createResetToken(20);
+            $mail = new PHPMailer();
+
+            $mail->SMTPDebug = SMTP::DEBUG_SERVER;
+            $mail->isSMTP();
+            $mail->Host       = $smtpConfig['host'];
+            $mail->SMTPAuth   = $smtpConfig['auth'];
+            $mail->Username   = $smtpConfig['username'];
+            $mail->Password   = $smtpConfig['password'];
+            $mail->SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS;
+            $mail->Port       = $smtpConfig['port'];
+
+            //Recipients
+            $mail->setFrom($smtpConfig['username'], $smtpConfig['user']);
+            $mail->addAddress($user->email, $user->name);
+
+            //Content
+            $mail->isHTML(true);
+            $mail->Subject = 'Reset Link';
+            $mail->Body    = '
+                <div style="max-width:600px;margin:0 auto;background-color:#ffffff;">
```

Comment:
Can you please isolate this in a separate file. something like Templates/Email/resestPassword.php, just include it here We can worry about template engine later, at least by that time all the templates will be in one place.

Reply:
Changes added!